functions.h

```diff
@@ -324,10 +315,10 @@ auto bessel_i_helper(const T& z) {
         y_tmp = 2. * n / z * y0 + y1;
         y1 = y0;
         y0 = y_tmp;
-        mu += 2. * (n % 2 == 0 ? 1 : -1) * (z.real() < 0 ? -1 : 1) * y1;
+        mu += 2. * (std::real(z) < 0 ? 1 - 2 * (n & 1) : 1) * y1;
     }
 
-    mu = std::exp(z.real() < 0 ? -z : z) * (mu + y0);
+    mu = std::exp(std::real(z) < 0 ? z : -z) * (mu + y0);
```

Comment:
It seems that the negative sign can change only change the world just a little bit.